src/js/service/restService.js

```diff
@@ -0,0 +1,68 @@
+import { util } from '../util/util';
+import {GridActionREST} from "../model/GridActionREST";
+
+let restService = {};
+
+restService.doAction = async function (action) {
+    try {
+        //ensure to have default values also in case the user deletes the field contents
+        let method=action.method;
+        if(!method) {
+            method=GridActionREST.defaults.method;
+        }
+
+        let contentType=action.contentType;
+        if(contentType === '') {
+            contentType=GridActionREST.defaults.contentType;
+        }
+
+        //as method GET does not have a body, conditionally set the body variable
+        let body;
+        if(method!='GET') {
+            body=action.body;
+        }
+
+        console.log(`url: ${action.restUrl}, body: ${body}, method: ${method}, contenttype: ${contentType}`);
+        const response = await fetch(action.restUrl, {
+                method: method,
+                //mode: "no-cors",
+                headers: {
+                    'Content-Type': contentType
+                },
+                body: body
+            }
+        );
+        if(!response.ok) {
+            console.error(`REST call failed with status message (${response.statusText}), statusCode (${response.status})`);
+        } else {
+            //console.log(`response body: ${response.json().then()}`);
+            log.debug(`REST call ok, url: ${action.restUrl}, body ${action.body}`)
+        }
+    }catch(error) {
+        console.error(error);
+    }
+};
+
+restService.getRestURL = function (userUri) {
```

Comment:
I think there should be something like a generic `util.getRestURL(userUrl, options)` which then is used here, and at `openHABService.getRestURL()` and `areService.getRestURL()`

Reply:
will remove this method as it's not suitable for a generic REST method as there are too many variables.